src/core/config/type-cast/type-casters/number.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * @prettier
+ */
+const numberTypeCaster = (value) => {
```

Comment:
This function can now return 3 different types:

```js
numberTypeCaster(NaN) => NaN
numberTypeCaster("string") => string
numberTypeCaster("1") => 1
```

technically `typeof NaN === typeof 1`, so the function only returns two types, but you have to approach `NaN` in exception way.

I would recommend this function to return on `typeof x === "number"`, which means it should either return real number or NaN. NaN is comparable to real numbers  in a way that it always returns false.